sdks/java/core/src/main/java/org/apache/beam/sdk/util/TimerInternals.java

```diff
@@ -164,16 +163,31 @@
    * Data about a timer as represented within {@link TimerInternals}.
    */
   class TimerData implements Comparable<TimerData> {
+    private final String timerId;
     private final StateNamespace namespace;
     private final Instant timestamp;
     private final TimeDomain domain;
 
-    private TimerData(StateNamespace namespace, Instant timestamp, TimeDomain domain) {
+    private TimerData(
+        StateNamespace namespace, String timerId, Instant timestamp, TimeDomain domain) {
+      this.timerId = timerId;
```

Comment:
Order here should match order of parameters and order of fields?

`@AutoValue`?